assets/validation.md

```diff
@@ -0,0 +1,80 @@
+<!---
+@startuml
+start
+#lightgreen:Order arrives via MQTT;
+#lightgreen:Validate JSON;
+if ((1) is received order valid?) then (no)
+    #red:reject order
+    throw error;
+    stop
+endif
+->yes;
+    if ((2) is received order new?) then (yes)
+        if ((3) is vehicle still executing\nan order or waiting for an update?) then (yes)
+        #red:reject order\nthrow error;
+         stop
+       endif
+       ->no;
+            if ((4) is start of new order close\nenough to current position?) then (no)
+                #red:reject order\nthrow error;
+                stop
+            endif
+            ->yes;
+                #lightgreen:delete states of previous order;
+                #lightgreen:accept order\nset orderId and orderUpdateId\npopulate states of new order (9);
+          #lightgreen: execute order;
+     else (no - received order is update of current order)
+        if ((5) is received order update deprecated?) then (yes)
+            #red:reject order\nthrow error;
```

Comment:
Why throwing an error here? IMHO this is the same case as "(6) is order update currently on vehicle" (n-1 vs n-t). 

A outdated order update causes no harm. The AMR has processed it already and can simply continue its current task. The operator should not be bothered with errors like: "We received an old order update but everything else is fine, nothing to worry about..." Additionaly this would simplify the ControlFlow.

And if (cause of programming errors) the MC sends outdated OrderMessages "on purpose" this should easily be detectable via comparison of State/Order Topic.

Logical reasoning: If a system is allowed to (or cannot prevent to) send a OrderUpdate multiple times (case 6) than the same system cannot prevent to "send" different OrderUpdates "at the same time". With HA deployments and "different paths" a OrderUpdate n-1 can be still "enroute" while OrderUpdate n overtakes it ...

Reply:
This PR is against version 2.0.1 and therefor shall not change anything. It is intended with this PR to clarify the current specified validation process, not to change it. The VDA / VDMA were strongly encouraged, to avoid breaking changes in the VDA5050 for the next future.

Reply:
@collani-bosch Is there a changelog or discussion log of issues like this that should be revisited between different version updates? Ex 2.0.1 vs 2.1 vs 3.0?
I for one agree with @raptor2101 on this, as a given network's race conditions and packet routing conditions breaks communication flow, which partially defeats the benefit of MQTT's unreliable network transfer protocol.

Reply:
@JWCS : I think the readme in each branch describes whats the goal for the specific version, but there is no changelog (as far as I know). You can help yourself by comparing the markdown.